pandas/core/groupby/ops.py

```diff
@@ -780,7 +780,11 @@ def get_iterator(self, data: FrameOrSeries, axis: int = 0):
         Generator yielding sequence of (name, subsetted object)
         for each group
         """
-        slicer = lambda start, edge: data._slice(slice(start, edge), axis=axis)
+        if axis == 0:
```

Comment:
a more natural idiom (used in many places is)
```
indexer = [None] * 2
indexer[axis] = slice(start, edge)
slicer = data.iloc[indexer]
```

Reply:
i guess, but thats not really conducive to being a lambda

Reply:
why does this need to be a lamba?

Reply:
doesnt have to be, thats just the pattern we have now.  i find `iloc[:, start:end]` clearer than the 3-line version

Reply:
ok